### ISSUE TITLE
[no-nested-interactive] false positive with content within {{#if}}/{{else}} – apply similar fix as with no-duplicate-id?

### ISSUE THREAD

**Issue by johanrd** [ember-template-lint/ember-template-lint#3159]
The no-nested-interactive rule in ember-template-lint currently flags conditionally rendered interactive elements as errors, even when only one instance of the element will be present in the DOM at any given time. This is a false positive, similar to the issue that was previously addressed for the no-duplicate-ids here: https://github.com/ember-template-lint/ember-template-lint/issues/1511

## Reproduction Case
Consider the following Glimmer component template:

```hbs
<label> My input:
  {{#if @select}}
    <select></select>
  {{else}}
    <input type='text'>
  {{/if}}
</label>
```
In this example, there are two elements inside the label, but they are mutually exclusive. 

## Proposed solution
This general concept seems to have been fixed in no-duplicate-id, see: https://github.com/ember-template-lint/ember-template-lint/pull/1606
